.github/workflows/ci.yml

```diff
@@ -0,0 +1,305 @@
+          toolchain: ${{ matrix.rust }}
+          target: ${{ matrix.target }}
+          profile: minimal
+          components: rustfmt
+          default: true
+          override: true
+      - name: Cargo cache
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/.cargo/registry
+            ~/.cargo/git
+          key: ${{ runner.os }}-cargo-rust_${{ matrix.rust }}-${{ hashFiles('**/Cargo.toml') }}
+      - name: Build cache
+        uses: actions/cache@v2
+        with:
+          path: target
+          key: ${{ runner.os }}-build-rust_${{ matrix.rust }}-target_${{ matrix.target }}-${{ hashFiles('**/Cargo.toml') }}
+      - name: Update deps
+        uses: actions-rs/cargo@v1
+        with:
+          command: update
+      - name: Build sys
+        uses: actions-rs/cargo@v1
+        env:
+          RUST_LOG: bindgen=warn,bindgen::ir=error,bindgen::codegen=error
+        with:
+          command: build
+          args: --manifest-path drm-ffi/drm-sys/Cargo.toml --target ${{ matrix.target }} --features update_bindings
+      - name: Upload bindings
+        if: matrix.task == 'bindings'
+        uses: actions/upload-artifact@v2
+        with:
+          name: bindings
+          path: |
+            drm-ffi/drm-sys/${{ env.DRM_SYS_BINDINGS_FILE }}
+            LICENSE
+            README.md
+      - name: Build
+        uses: actions-rs/cargo@v1
+        with:
+          command: build
+          args: --target ${{ matrix.target }}
+      - name: Test
+        if: contains(matrix.target, '-linux-') && (startsWith(matrix.target, 'x86_64-') || startsWith(matrix.target, 'i686-'))
+        uses: actions-rs/cargo@v1
+        timeout-minutes: 12
+        env:
+          RUST_BACKTRACE: full
+        with:
+          command: test
+          args: --all --target ${{ matrix.target }}
+
+  update-bindings:
+    if: ${{ github.event_name != 'pull_request' && !startsWith(github.ref, 'refs/tags/') }}
+    needs:
+      - test
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Download bindings
+        uses: actions/download-artifact@v2
+        with:
+          name: bindings
+      - name: Create pull request
+        uses: peter-evans/create-pull-request@v3
+        with:
+          base: ${{ github.head_ref }}
+          commit-message: Updated bindings
+          branch: update-bindings
+          delete-branch: true
+          title: Update bindings
+          body: |
+            Bindings should be updated to be consistent with latest changes
+
+  publish:
+    if: github.repository == 'Smithay/drm-rs' && startsWith(github.ref, 'refs/tags/v')
+    needs:
+      - format
+      - doc
+      - check
+      - check-minimal
+      - test
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Setup Rust
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: stable
+          override: true
+      - name: Publish crates
```

Comment:
Can this step deal with the three different packages of the repository?
Or do we need multiple steps here?

The order is important as well, since the crates depend on each other.
We need to push `drm-sys` first, then `drm-ffi` and at last `drm`.

Reply:
This step uses [publish-crates](https://github.com/marketplace/actions/publish-crates) action which supports publishing workspace members with dependencies.
It works for my projects just a fine. I hope it will also works here.